README.md
# terranean wp starter
<em>terranean starter theme originally based on [Air Light](https://github.com/digitoimistodude/starter-terranean-wp)</em>


## Requirements:
- Node version : 18.14.2

## Additions:
- [TailwindCSS](https://tailwindcss.com/)
- [Slick](https://kenwheeler.github.io/slick/)
- [Preline Components](https://preline.co/index.html)

## How-To:

### Add Custom Javascript
1. Add module file to /js/src/terranean
2. Import module and call it in /js/src/front-end.js (or whichever is applicable)

### Work With Preline Components
## Adding JS for each component
1. Look at list of module exports in /js/src/modules/terranean/preline-src/index
2. Import necessary modules and call them in /js/src/front-end.js (or whichever is applicable)

## Issues:
- gulp watch does not work

CHANGELOG.md
### [Unreleased]

* Reset core-list bullet size
* Remove un-used `image_sizes` array from theme settings
* Run `theme_setup` on `after_setup_theme` hook
* Add custom `pll_translatable` setting to cpt and tax registration, use that to automatically register that content type for Polylang
* Fix ACF button color
* Remove deprecated number-leading-zero rule

### 9.3.2: 2023-03-21

* When page is loaded with "s" in url parameters try to set focus to search input field (kudos to @EliasKau)
* Calculate mobile nav top and height in js when there are air notificans present (kudos to @Tumppex)
* Fix irregularities with keyboard navigation #175 (kudos to @michaelbourne!)
* Fix navigation focus trap #175 (kudos to @michaelbourne!)
* Stylelint: Add number-leading-zero from devpackages
* Fix navigation issues #177 (kudos to @michaelbourne!)

### 9.3.1: 2023-02-24

* Remove nav-toggle styles not in use
* Add air-notification support to navigation.js
* Simplify button component even more by removing unused arrow conditional

### 9.3.0: 2023-02-17

* Change nav toggle position to static by default (kudos to @Tumppex)
* Add wrapper to navigation (kudos to @Tumppex)
* Mobile nav variable to change nav width (kudos to @Tumppex)

### 9.2.9: 2023-02-15

* Start script: Copy .nvmrc to project root from devpackages
* Improve block cache bypass logic and show bypass reason on log
* Try to determine if block has Gravity Forms form and always bypass cache if 
* Stylelint fixes
* Simplify button SCSS component
* Remove Travis
* Add GitHub workflows for styles, html, php and js
* Fix version in .nvmrc
* Add badges for GitHub workflow build statuses
* Fix stylelint errors for v15
* Add smaller font-sizes 12 and 13
* Replace all suitable px sizes with rem as per stylelint-rem-over-px
* Update .stylelintrc from devpackages 2.5.3

### 9.2.8: 2023-01-30

* Add gap for categories
* Change default mobile breakpoint to 600px
* Remove breakpoints not in use
* Fix iPad navigation overlapping blocks when closed

### 9.2.7: 2023-01-06

* Remove unused no-js styles
* Remove unused CSS variables
* Simplify defining CSS variables in single use cases
* Remove `<kbd>` styles that are rarely needed
* Fix entry-footer
* Simplify category styles
* Simplify tag styles
* Remove _blog.scss
* Add views/_single.scss
* Use list element in categories instead of a paragraph

### 9.2.6: 2022-12-31

* Fix undefined body class
* Fix nav styles leaking to pagination
* Drop sanitize.css dependency
* Use sanitize.css/reduce-motion baked-in instead from npm

### 9.2.5: 2022-12-31

* Major navigation.js rewrite
* Modularize navigation.js
* Make navigation dependency free, completely dropping jQuery
* Perfected arrow and keyboard navigation patterns for A11y
* Fix aria-labels and aria-expanded
* Move nav-toggle inside nav landmark
* Fix focus trap
* Rewrite nav-toggle and 🍔
* Use clever way to visually present attr(aria-label) for 🍔
* Define mobile breakpoint more modern way with getPropertyValue
* Drop IE10-IE11 support for navigation.js
* Drop hamburgers npm module dependency
* Fix focusable elements if last item is a button
* Fix checkForSubmenuOverflow() function
* Fix resize functions
* Remove useless vars and functions
* Add plus and minus icons for mobile as default as inlined SVG
* Drop devDependencies from theme package.json, add instructions on contributing
* Fix acf-button color inside a Gutenberg block
* Navigation.js: Bail if navigation doesn't exist
* Remove leftover deprecated gutenberg-helpers.js
* Allow :not notations where needed

### 9.2.4: 2022-12-02

* Upgrade to [devpackages 2.4.9](https://github.com/digitoimistodude/devpackages/releases/tag/2.4.9)
* Disable no-descending-specificity rule in stylelint
* Ignore some obvious phpcs nags

### 9.2.3: 2022-11-28

* Add core/list-item to allowed blocks due to WP 6.1 and latest Gutenberg editor
* Remove extraneous padding in the first block on Gutenberg editor
* Make heading CSS variable naming more consistent
* Button component: Do not use border-radius on ghost button as it's inherited from button
* Button component: Use own border radius instead of shared with form fields
* Update generated README.md in the start script
* Fix ACF relation field color in Gutenberg blocks
* Simplify _font-family.scss
* Deprecate core/code, core/columns, core/cover, core/embed, core/preformatted and code/verse CSS from terranean-wp-starter, use Gutenberg official styles instead
* Add Inter as variable font

### 9.2.2: 2022-11-04

* Fix incorrect textdomain in external-link JS module
* Simplify CPT and taxonomy registration by using the class name also as a slug (Merge pull request #158 from digitoimistodude/cpt-tax-register-simplification, kudos to @timiwahalahti)
* Remove what-input depenency, use :focus and :focus-within instead, remove forced focus outline-color
* Bump tested up WordPress version to 6.1

### 9.2.1: 2022-10-28

* Ensure all blocks get default alignment styles despite core styles or Autoptimize order
* Increase default $transition-duration

### 9.2.0: 2022-10-12

* Add check for a11ySkipLink
* Add missing external-link-icon class
* Fix aspect-ratio mixin supports query
* Fix mobile nav-toggle one side is longer than others while button focused

### 9.1.9: 2022-10-06

* Disable stylelint max-line-length
* Remove default list color conflicting with ACF fields
* Remove global link focus color that is no longer used
* Improve focus ring style, add variable for color: `--color-focus-outline`
* Improve accessible Finnish translation of back to top
* Improve back to top keyboard accessibility

### 9.1.8: 2022-09-29

* Remove leftover styles from default block
* Remove outdated scripts in gutenberg-editor.js that causes an error on wp-admin

### 9.1.7: 2022-09-28

* Add external-link-icon styles for mobile
* Remove setup_editor_styles function because it breaks stage+prod and block styles still load correctly without that
* Improve anchors.js formatting

functions.php
<?php
/**
 * Gather all bits and pieces together.
 * If you end up having multiple post types, taxonomies,
 * hooks and functions - please split those to their
 * own files under /inc and just require here.
 *
 * @Date: 2019-10-15 12:30:02
 * @Last Modified by:   Timi Wahalahti
 * @Last Modified time: 2023-03-31 14:50:38
 *
 * @package terranean-wp-starter
 */

namespace Air_Light;

/**
 * The current version of the theme.
 */
define( 'AIR_LIGHT_VERSION', '9.3.2' );

// We need to have some defaults as comments or empties so let's allow this:
// phpcs:disable Squiz.Commenting.InlineComment.SpacingBefore, WordPress.Arrays.ArrayDeclarationSpacing.SpaceInEmptyArray

/**
 * Theme settings
 */
add_action( 'after_setup_theme', function() {
  $theme_settings = [
    /**
     * Theme textdomain
     */
    'textdomain' => 'terranean-wp-starter',

    /**
     * Content width
     */
    'content_width' => 800,

    /**
     * Logo and featured image
     */
    'default_featured_image'  => null,
    'logo'                    => '/svg/logo.svg',

    /**
     * Custom setting group settings when using Air setting groups plugin.
     * On multilingual sites using Polylang, translations are handled automatically.
     */
    'custom_settings' => [
      // 'your-custom-setting' => [
      //   'id' => Your custom setting post id,
      //   'title' => 'Your custom setting',
      //   'block-editor' => true,
      //  ],
    ],

    'social_media_accounts'  => [
      // 'twitter' => [
      //   'title' => 'Twitter',
      //   'url'   => 'https://twitter.com/digitoimistodude',
      // ],
    ],

    /**
     * All links are cheked with JS, if those direct to external site and if,
     * indicator of that is included. Exclude domains from that check in this array.
     */
    'external_link_domains_exclude' => [
      'localhost:3000',
      'airdev.test',
      'airwptheme.com',
      'localhost',
    ],

    /**
     * Menu locations
     */
    'menu_locations' => [
      'primary' => __( 'Primary Menu', 'terranean-wp-starter' ),
    ],

    /**
     * Taxonomies
     *
     * See the instructions:
     * https://github.com/digitoimistodude/terranean-wp-starter#custom-taxonomies
     */
    'taxonomies' => [
      // 'Your_Taxonomy' => [ 'post', 'page' ],
    ],

    /**
     * Post types
     *
     * See the instructions:
     * https://github.com/digitoimistodude/terranean-wp-starter#custom-post-types
     */
    'post_types' => [
      // 'Your_Post_Type',
    ],

    /**
     * Gutenberg -related settings
     */
    // Register custom ACF Blocks
    'acf_blocks' => [
      // [
      //   'name'           => 'block-file-slug',
      //   'title'          => 'Block Visible Name',
      //   // You can safely remove lines below if you find no use for them
      //   'prevent_cache'  => false, // Defaults to false,
      //   // Icon defaults to svg file inside svg/block-icons named after the block name,
      //   // eg. svg/block-icons/block-file-slug.svg
      //   //
      //   // Icon setting defines the dashicon equivalent: https://developer.wordpress.org/resource/dashicons/#block-default
      //   // 'icon'  => 'block-default',
      // ],
    ],

    // Custom ACF block default settings
    'acf_block_defaults' => [
      'category'          => 'terranean-wp-starter',
      'mode'              => 'auto',
      'align'             => 'full',
      'post_types'        => [
        'page',
      ],
      'supports'  => [
        'align'           => false,
        'anchor'          => true,
        'customClassName' => false,
      ],
      'render_callback'   => __NAMESPACE__ . '\render_acf_block',
    ],

    // Restrict to only selected blocks
    // Set the value to 'all' to allow all blocks everywhere
  //  'allowed_blocks' => [
  //     'default' => [
  //       'all'
  //     ],
  //     'post' => [
  //       'core/archives',
  //       'core/audio',
  //       'core/buttons',
  //       'core/categories',
  //       'core/code',
  //       'core/column',
  //       'core/columns',
  //       'core/coverImage',
  //       'core/embed',
  //       'core/file',
  //       'core/freeform',
  //       'core/gallery',
  //       'core/heading',
  //       'core/html',
  //       'core/image',
  //       'core/latestComments',
  //       'core/latestPosts',
  //       'core/list',
  //       'core/list-item',
  //       'core/more',
  //       'core/nextpage',
  //       'core/paragraph',
  //       'core/preformatted',
  //       'core/pullquote',
  //       'core/quote',
  //       'core/block',
  //       'core/separator',
  //       'core/shortcode',
  //       'core/spacer',
  //       'core/subhead',
  //       'core/table',
  //       'core/textColumns',
  //       'core/verse',
  //       'core/video',
  //     ],
  //   ],

    // If you want to use classic editor somewhere, define it here
    'use_classic_editor' => [],

    // Add your own settings and use them wherever you need, for example THEME_SETTINGS['my_custom_setting']
    'my_custom_setting' => true,
  ];

  $theme_settings = apply_filters( 'air_light_theme_settings', $theme_settings );

  define( 'THEME_SETTINGS', $theme_settings );
} ); // end action after_setup_theme

/**
 * Required files
 */
require get_theme_file_path( '/inc/hooks.php' );
require get_theme_file_path( '/inc/includes.php' );
require get_theme_file_path( '/inc/template-tags.php' );

// Run theme setup
add_action( 'after_setup_theme', __NAMESPACE__ . '\theme_setup' );
add_action( 'after_setup_theme', __NAMESPACE__ . '\build_theme_support' );

// Blockstudio editor
add_filter('blockstudio/editor/users', function () {
  return [1];
});